### PR TITLE
fix: Site freezes after agent logs out of ASM

### PR DESCRIPTION
**Pull request by bullfroggy-sap**
Revert ["Site freezes when logging out of ASM"](https://github.com/SAP/spartacus/commit/10bf6c09ce691e1967d751eeb909f16dcef32cb3)
Explicitly set logout process back to false after the operation has completed
Create e2e and unit tests to verify new logout behavior

### CODE REVIEW

File: projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts [hunk 1]
```diff
@@ -8,6 +8,14 @@ import { getSampleUser } from '../../../sample-data/checkout-flow';
...
 import * as consent from '../../../helpers/consent-management';
 import { fillShippingAddress } from '../../../helpers/checkout-forms';
+import {
+  waitForCategoryPage,
+  waitForPage,
+} from '../../../helpers/checkout-flow';
+import {
+  navigateToCategory,
+  navigateToHomepage,
+} from '../../../helpers/vendor/cds/merchandising-carousel';
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts:16]: Could we extract `navigateToCategory` and `navigateToHomepage` in a more common helper than CDS (cds being an integration library)

Comment (bullfroggy-sap) [reply projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts:16]: I thought you might say that. Consider it done.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cds/merchandising-carousel.ts [hunk 2]
```diff
@@ -344,12 +344,13 @@ export function navigateToHomepage(): void {
 
 export function navigateToCategory(
   categoryName: string,
-  categoryCode: string
+  categoryCode: string,
+  force: boolean = true
```
Comment (plabadie) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cds/merchandising-carousel.ts]: Could we revert this changes now that we don't reuse the functions from this file?

Comment (bullfroggy-sap) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/vendor/cds/merchandising-carousel.ts]: hm yes i suppose so

**Review by plabadie — APPROVED** [review]
